Feature/BJS2-66792-unit-test

Zhltsk-V opened the pull request.

Commit:
Merge branch 'murloc-master-stream9' into feature/BJS2-66792-unit-test

Commit:
Merge branch 'feature/BJS2-65903-event-organization' into feature/BJS2-66792-unit-test

Commit:
Merge branch 'refs/heads/feature/BJS2-65903-event-organization' into feature/BJS2-66792-unit-test

Commit:
The task BJS2-66792 has been completed.

- Bugs in EventService have been fixed and documentation has been updated.
- Unit tests for EventService and EventFilter have been added.

marsel-mkh commented on src/test/java/school/faang/user_service/service/filter/event/EventEndDateFilterTest.java:
```diff
@@ -0,0 +1,86 @@
+    @Test
+    @DisplayName("Тест apply: фильтрация событий с endDate, который позже указанной даты")
+    void applyTestAfterEndDate() {
+        Event event = Event.builder().endDate(LocalDateTime.parse("2022-12-12T12:12:12")).build();
```
Разграничивай "AAA" пробелами

marsel-mkh commented on src/test/java/school/faang/user_service/service/filter/event/EventLocationFilterTest.java:
```diff
@@ -0,0 +1,97 @@
+    @Test
+    @DisplayName("Тест isApplicable: фильтр применим, если location установлен и не пустой")
+    void testIsApplicableTestTrue() {
+        eventFilterDto.setLocation("Moscow");
```
Тут тоже разгначь все пробелами,будет читабельнее

marsel-mkh commented on src/test/java/school/faang/user_service/service/filter/event/EventStartDateFilterTest.java:
```diff
@@ -0,0 +1,68 @@
+    @Test
+    @DisplayName("Тест isApplicable: фильтр применим, если startDate установлен")
+    void isApplicableTestTrue() {
+        boolean result = eventStartDateFilter.isApplicable(eventFilterDto);
```
Пробелы

marsel-mkh commented on src/test/java/school/faang/user_service/service/filter/event/EventEndDateFilterTest.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    @Test
+    @DisplayName("Тест isApplicable: фильтр применим, если endDate установлен")
```
перепиши названия в стиле "Проверка, что фильтр применим, если endDate установлен"

marsel-mkh commented on src/test/java/school/faang/user_service/service/service/event/EventServiceTest.java:
```diff
@@ -0,0 +1,465 @@
+
+@ExtendWith(MockitoExtension.class)
+public class EventServiceTest {
+
```
Лишний пробел

marsel-mkh commented on src/test/java/school/faang/user_service/service/service/event/EventServiceTest.java:
```diff
@@ -0,0 +1,465 @@
+        when(userRepository.findById(eventCreateDto.getOwnerId())).thenReturn(Optional.of(eventOwner));
+        when(skillRepository.findAllById(requiredSkillsIds)).thenReturn(requiredSkills);
+
+        assertThrows(NullPointerException.class, () -> eventService.create(eventCreateDto));
```
Не совсем понятно почему npe

Commit:
Task BJS2-66792.
- Changes have been made to the task.

RinatBeybutov commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -83,7 +82,8 @@ public List<EventViewDto> getEventsByFilter(EventFilterDto eventFilterDto) {
 
-        for (EventFilter eventFilter : eventFilters) {
+        for (int i = 0; i < eventFilters.size(); i++) {
+            var eventFilter = eventFilters.get(i);
```
для чего здесь был изменен цикл?

Zhltsk-V replied:
Из-за тестов на фильтры. При тестировании тут всегда выбрасывало NPE, т.к. я неправильно делал moсk объект листа фильтров (если помнишь, проблему рассказывал на ретро). 
Сейчас можно вернуть назад)

RinatBeybutov replied:
давай вернем обратно)

RinatBeybutov commented on src/test/java/school/faang/user_service/service/filter/event/EventEndDateFilterTest.java:
```diff
@@ -0,0 +1,99 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class EventEndDateFilterTest {
```
Давай добавим над классом тоже 
@DisplayName("Проверка фильтрации ..")
А над классом еще можно java docs добавить:
```
/**
* Тесты для проверки фильтрации сущности  {@link ...}
*/
```

RinatBeybutov commented on src/test/java/school/faang/user_service/service/filter/event/EventEndDateFilterTest.java:
```diff
@@ -0,0 +1,99 @@
+
+    @BeforeEach
+    void setUp() {
+        eventFilterDto = new EventFilterDto();
```
давай напишем хотя бы один тест с вызовом метода сервиса, там должен быть мок репозитория, который возвращает несколько объектов, а в итоге из сервиса получаем отфильтрованные данные. В других тестовых классах тоже добавь такой метод

Commit:
In Task BJS2-6679, the following fixes and improvements were made:

- Added tests to EventEndDateFilterTest, EventLocationFilterTest, EventStartDateFilterTest, EventTitleFilterTest, and EventServiceTest.
- Added/updated JavaDoc documentation.
- Added endpoints in the EventController class.
- Modified the logic of the updateEvent method in the EventController class.
- Added a new exception.
- Fixed minor bugs and issues.
- Added the getEventById method to the EventService class.
- Added a new updateEntity method to the EventMapper.

RinatBeybutov commented on src/test/java/school/faang/user_service/service/filter/event/EventEndDateFilterTest.java:
```diff
@@ -96,4 +133,53 @@ void applyTestAfterEndDate() {
+        when(eventMapper.toDto(event3)).thenReturn(new EventViewDto());
+        when(eventMapper.toDto(event4)).thenReturn(new EventViewDto());
+
+        when(eventEndDateFilter.isApplicable(filter)).thenReturn(true);
```
а зачем мокать фильтр? мы же как раз его работу и проверяем

RinatBeybutov replied:
убрать этот мок для фильтра и все будет работать корректно

RinatBeybutov commented on src/test/java/school/faang/user_service/service/filter/event/EventEndDateFilterTest.java:
```diff
@@ -96,4 +133,53 @@ void applyTestAfterEndDate() {
+                .endDate(LocalDateTime.parse("2021-10-10T10:10:10"))
+                .build();
+
+        EventEndDateFilter eventEndDateFilter = Mockito.mock(EventEndDateFilter.class);
```
здесь нужно не mock а new EventEndDateFilter()

Commit:
Merge branch 'murloc-master-stream9' into feature/BJS2-66792-unit-test

Commit:
Updates have been added to task BJS2-66792.
